src/viam/sdk/CMakeLists.txt

```diff
@@ -0,0 +1,204 @@
+    # be initialized within main before anything else happens?
+    registry/registry.cpp
+    common/proto_type.cpp
+    common/utils.cpp
+    components/camera/camera.cpp
+    components/camera/client.cpp
+    components/camera/server.cpp
+    components/component_base.cpp
+    components/generic/client.cpp
+    components/generic/generic.cpp
+    components/generic/server.cpp
+    components/motor/motor.cpp
+    components/motor/client.cpp
+    components/motor/server.cpp
+    config/resource.cpp
+    module/handler_map.cpp
+    module/module.cpp
+    module/service.cpp
+    referenceframe/frame.cpp
+    resource/resource.cpp
+    resource/resource_base.cpp
+    resource/resource_manager.cpp
+    resource/resource_server_base.cpp
+    resource/resource_type.cpp
+    robot/client.cpp
+    rpc/dial.cpp
+    rpc/server.cpp
+    services/service_base.cpp
+    spatialmath/geometry.cpp
+    spatialmath/orientation_types.cpp
+    spatialmath/orientation.cpp
+    subtype/subtype.cpp
+    tests/test_utils.cpp
+    tests/mocks/camera_mocks.cpp
+    tests/mocks/generic_mocks.cpp
+    tests/mocks/mock_motor.cpp
+  PUBLIC FILE_SET viamsdk_public_includes TYPE HEADERS
+    BASE_DIRS
+      ../..
+    FILES
+      ../../viam/sdk/common/proto_type.hpp
+      ../../viam/sdk/common/utils.hpp
+      ../../viam/sdk/components/camera/camera.hpp
+      ../../viam/sdk/components/camera/client.hpp
+      ../../viam/sdk/components/camera/server.hpp
+      ../../viam/sdk/components/component_base.hpp
+      ../../viam/sdk/components/generic/client.hpp
+      ../../viam/sdk/components/generic/generic.hpp
+      ../../viam/sdk/components/generic/server.hpp
+      ../../viam/sdk/components/motor/motor.hpp
+      ../../viam/sdk/components/motor/client.hpp
+      ../../viam/sdk/components/motor/server.hpp
+      ../../viam/sdk/components/service_base.hpp
+      ../../viam/sdk/config/resource.hpp
+      ../../viam/sdk/module/handler_map.hpp
+      ../../viam/sdk/module/module.hpp
+      ../../viam/sdk/module/service.hpp
+      ../../viam/sdk/referenceframe/frame.hpp
+      ../../viam/sdk/registry/registry.hpp
+      ../../viam/sdk/resource/resource.hpp
+      ../../viam/sdk/resource/resource_base.hpp
+      ../../viam/sdk/resource/resource_manager.hpp
+      ../../viam/sdk/resource/resource_server_base.hpp
+      ../../viam/sdk/resource/resource_type.hpp
+      ../../viam/sdk/robot/client.hpp
+      ../../viam/sdk/robot/service.hpp
+      ../../viam/sdk/rpc/dial.hpp
+      ../../viam/sdk/rpc/server.hpp
+      ../../viam/sdk/services/service_base.hpp
+      ../../viam/sdk/spatialmath/geometry.hpp
+      ../../viam/sdk/spatialmath/orientation_types.hpp
+      ../../viam/sdk/spatialmath/orientation.hpp
+      ../../viam/sdk/subtype/subtype.hpp
+      ../../viam/sdk/tests/test_utils.hpp
+      ../../viam/sdk/tests/mocks/camera_mocks.hpp
+      ../../viam/sdk/tests/mocks/generic_mocks.hpp
+      ../../viam/sdk/tests/mocks/mock_motor.hpp
+)
```

Comment:
Same fileset change here. Now the fileset understands the relative pathing expected at build and install time and does the right thing automatically, and also doesn't inject a spurious (and potentially shadow-y) path that would allow inclusions without the proper `<viam/[app|sdk` prefix.

Reply:
[q] `../../viam/sdk/component/*.hpp` is preferred over `./component/*.hpp` because otherwise there is the potential for unintentional path inclusions? Wouldn't `./component/*.hpp` already ensure that the path starts with `viam/sdk` because this CMakeLists file is in that directory?

Reply:
And, having said that, I realized that it probably means we can remove the manual setting for the install-time target include directories for libviamapi and libviamsdk because the FILE_SET takes care of that too. Doing that.